hw/dv/dpi/usbdpi/usbdpi.c

```diff
@@ -529,48 +512,40 @@ void setBaud(struct usbdpi_ctx *ctx) {
   }
 }
 
-// Test the ischronous transfers (without ACKs)
-void testIso(struct usbdpi_ctx *ctx) {
+// Try OUT transfer to the device, optionally expecting a Status
+//   packet (eg. ACK|NAK) in response; this is not expected for
+//   Isochronous transfers
+void tryTX(usbdpi_ctx_t *ctx, bool bExpectStatus) {
+  usbdpi_transfer_t *tr = ctx->sending;
+  uint8_t *dp;
   switch (ctx->hostSt) {
     case HS_STARTFRAME:
-      ctx->state = ST_SYNC;
-      ctx->bytes = 14;
-      ctx->datastart = 3;
-      ctx->byte = 0;
-      ctx->bit = 1;
-      ctx->data[0] = USB_PID_OUT;
-      ctx->data[1] = 0x82;
-      ctx->data[2] = 1 | CRC5(0x182, 11) << 3;
-
-      ctx->data[3] = USB_PID_DATA0;
-      ctx->data[4] = 0x42;  // h2d, vendor, endpoint
-      ctx->data[5] = 3;     // set baud
-      ctx->data[6] = 96;    // index 0
-      ctx->data[7] = 0;     // type device
-      ctx->data[8] = 0;     // wIndex = 0
-      ctx->data[9] = 0;
-      ctx->data[10] = 0;  // wLength = 0
-      ctx->data[11] = 0;
-
-      add_crc16(ctx->data, ctx->datastart, 12);
-      // ctx->data[12] = 0x00; // pre-computed CRC32
-      // ctx->data[13] = 0xBD;
+      transfer_token(tr, USB_PID_OUT, ctx->dev_address, ENDPOINT_ISOCHRONOUS);
+
+      // TODO - DATA toggle synchronization
```

Comment:
Note: Isochronous endpoints should only ever use DATA0 PIDs. There's no toggle synchronization here.